File: C#Part2/HomeworkStringsAndTextProcessing/CorrectBrackets/CorrectBrackets.cs
//Write a program to check if in a given expression the brackets are put correctly.

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

class CorrectBrackets
{
    static void Main()
    {
        Console.Write("Enter the expresion: ");
        string input = Console.ReadLine();

        bool result = CheckBrakets(input);

        if (result)
        {
            Console.WriteLine("The expression is corect!");
        }
        else
        {
            Console.WriteLine("The expression is not corect!");
        }
    }

    private static bool CheckBrakets(string input)
    {
        int count = 0;

        for (int i = 0; i < input.Length; i++)
        {
            if (input[i] == '(')
            {
                count++;
            }
            if (input[i] == ')')
            {
                count--;
            }
            if (count < 0)
            {
                return false;
            }
        }
        if (count == 0)
        {
            return true;
        }
        else
        {
            return false;
        }
    
    }

}

File: 6 JavaScriptBasic/proverka na domashni/StringsJS/02. Correct brackets/Correct brackets.js
﻿// Write a JavaScript function to check if in a given expression the brackets are put correctly.

function checkBrackets(expression) {
    var stack = 0
      , i
    for (i = 0; i < expression.length && stack >= 0; i++) {
        if (expression.charAt(i) === '(') stack++;
        if (expression.charAt(i) === ')') stack--;
    }
    return !stack;
}
console.log(checkBrackets('((a+b)/5-d)'));
console.log(checkBrackets(')(a+b)).'));